File: Array&Hasning/1.ContainsDuplicate/ContainsDuplicate.py
from typing import List

def containsDuplicate(nums: List[int]) -> bool:
    seenNums = set()
    for num in nums:
        if num in seenNums:
            return True
        seenNums.add(num)
    return False


#def containsDuplicate(nums: List[int]) -> bool:
# return len(set(nums)) < len(nums)

print(containsDuplicate([1,1,1,3,3,4,3,2,4,2]))

File: Array&Hasning/1.ContainsDuplicate/ContainsDuplicate.cs
static bool hasDuplicate(int[] nums)
{
        HashSet<int> seenNums = [];
        foreach(int num in nums)
            if (!seenNums.Add(num))
                return true;
        return false;
}
/*static bool hasDuplicate(int[] nums)
    return new HashSet<int>(nums).Count < nums.Length;    
*/

Console.WriteLine(hasDuplicate([1,1,1,3,3,4,3,2,4,2]));

